### PR TITLE
fix ansible/ansible#54223

### LINKED ISSUE #54223: TypeError: unhashable type: 'list' in cloudflare_dns module (from extra comma in argument_spec)
##### SUMMARY
works in 2.7.9 but not in 2.8.0dev0. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

modules/net_tools/cloudflare_dns.py
##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0
  config file = /home/mz0/do3/ansible.cfg
  configured module search path = [u'/home/mz0/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /shr/bemigot/do2/lib/ansible
  executable location = /shr/bemigot/do2/bin/ansible
  python version = 2.7.15+ (default, Oct  2 2018, 22:12:08) [GCC 8.2.0]
```

##### CONFIGURATION

```paste below
RETRY_FILES_ENABLED(/home/mz0/do3/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Same behavior when running with (virtualenv) Python 3.7

##### STEPS TO REPRODUCE

Any playbook will fail i suppose. The problem is internal to this module and is easy to fix.

```yaml

```

##### ACTUAL RESULTS

```paste below

```

### PR DESCRIPTION
**Pull request by mz0**
##### SUMMARY

Fixes #54223 by dropping gratuitous comma in argument_spec

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
cloudflare_dns

### CODE REVIEW

**Comment by andreaso** [comment]
Thanks for the bug report as well as this PR. I'm going away over the weekend, but I'll take a closer look at the suggested fix during next week.

File: lib/ansible/modules/net_tools/cloudflare_dns.py [hunk 1]
```diff
@@ -794,7 +794,7 @@ def main():
             ('state', 'absent', ['record']),
             ('type', 'SRV', ['proto', 'service']),
             ('type', 'TLSA', ['proto', 'port']),
-        ],
+        ]
```
Comment (Akasurde) [inline lib/ansible/modules/net_tools/cloudflare_dns.py]: How about - 

```python
required_if=[
            ('state', 'present', ['record', 'type', 'value']),
            ('state', 'absent', ['record']),
            ('type', 'SRV', ['proto', 'service']),
            ('type', 'TLSA', ['proto', 'port']),
        ],
```

Comment (mz0) [reply lib/ansible/modules/net_tools/cloudflare_dns.py]: that's even better! Thanks @Akasurde

**Comment by Akasurde** [comment]
@mz0 Thanks for the contribution. @andreaso Thanks for the review.

**Pull request merged**